File: Dante1/Module1.07/Zadanie7.32/main.c
#include <stdio.h>

double km2nm (double distance);
double km2lm (double distance);

int main()
{
    double number;
    printf("Enter: ");

    int check = scanf("%lf", &number);
    if(check != 1)
    {
        printf("Incorrect input");
        return 1;
    }
    if(number < 0)
    {
        printf("Incorrect input data");
        return 2;
    }

    printf("%.3f\n%.3f", km2nm(number), km2lm(number));
    return 0;
}

double km2nm (double distance)
{
    if(distance < 0)
    {
        return -1;
    }
    return distance * 0.540000;
}

double km2lm (double distance)
{
    if(distance < 0)
    {
        return -1;
    }
    return distance * 0.621371;
}

File: Dante1/Module1.07/Zadanie7.32/main.rs
fn km2nm(dist: f64) -> f64 {
    if dist < 0.0 {
        return -1.0;
    }
    dist * 0.540000
}

fn km2lm(dist: f64) -> f64 {
    if dist < 0.0 {
        return -1.0;
    }
    dist * 0.621371
}

fn main() {
    let mut input: String = String::new();

    std::io::stdin().read_line(&mut input).expect("Incorrect");
    let dist: f64 = input.trim().parse().expect("error");

    if dist < 0.0 {
        panic!("Incorrect input data");
    }

    println!("{:.3} {:.3}", km2nm(dist), km2lm(dist))
}
